### PR TITLE
check status code and retry the deb package download

### PR DESCRIPTION
**Pull request by kmdkuk**
#1738 

There is no clear intention for the 10 of retry count and 10s of retry intervals.

Signed-off-by: kouki <kouworld0123@gmail.com>

### CODE REVIEW

File: worker/deb.go [hunk 1]
```diff
@@ -24,19 +27,42 @@ func InstallDebianPackage(ctx context.Context, client *http.Client, ghClient *ht
...
+			}
+			defer resp.Body.Close()
+			if !(resp.StatusCode >= 200 && resp.StatusCode < 300) {
+				return fmt.Errorf("status code %d is not a success", resp.StatusCode)
+			}
+			_, err = io.Copy(debPkg, resp.Body)
+			if err != nil {
+				return err
+			}
+			return nil
```
Comment (ymmt2005) [inline worker/deb.go:46]:
```suggestion
			return err
```

File: worker/deb.go [hunk 2]
```diff
@@ -24,19 +27,42 @@ func InstallDebianPackage(ctx context.Context, client *http.Client, ghClient *ht
...
+		func(ctx context.Context) error {
+			resp, err := client.Get(downloadURL)
+			if err != nil {
+				return err
+			}
+			defer resp.Body.Close()
+			if !(resp.StatusCode >= 200 && resp.StatusCode < 300) {
+				return fmt.Errorf("status code %d is not a success", resp.StatusCode)
+			}
+			_, err = io.Copy(debPkg, resp.Body)
```
Comment (ymmt2005) [inline worker/deb.go:42]: What happens if `io.Copy` fails (due to network error or something) then this function is retried?

I'm worrying about there would be malformed data after/before the complete deb data.

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

File: worker/deb.go (hunk 1 above)
Comment (kmdkuk) [reply worker/deb.go]: thank you! fix it.

File: worker/deb.go (hunk 2 above)
Comment (kmdkuk) [reply worker/deb.go:45]: I think you are right in your worrying.
I fixed it to do `Reset()` before `io.Copy`.

Comment (ymmt2005) [reply worker/deb.go:45]: Or you can use `io.ReadAll()` and store the result data, like this.

```go
var data []byte
... (snip)
    data, err = io.ReadAll(resp.Body)
    if err != nil { return err }
```

Comment (kmdkuk) [reply worker/deb.go:45]: That's going to be simpler.

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**